praw/models/reddit/subreddit.py

```diff
@@ -1487,6 +1498,19 @@ def update(
             "text_color": text_color,
             "text_editable": text_editable,
         }
+        if fetch:
+            _existing_data = [
+                template
+                for template in iter(self)
+                if template["id"] == template_id
+            ]
+            if len(_existing_data) != 1:
+                raise InvalidFlairTemplateID(template_id)
+            else:
+                existing_data = _existing_data[0]
+                for key, value in existing_data.items():
+                    if data.get(key, "DoesNotExistInData") is None:
```

Comment:
What's the point of the "DoesNotExistInData" sentinel value? If this line were instead `if data.get(key) is None:`, then any key that doesn't exist *or* is `None` in `data` would be set by the existing data. With your change, you seem to only want to set items in `data` from `existing_data` if the key already exists in `data`.

To me, it seems like you made a deliberate choice to not set any values to `data` that are not already keys in `data`. Why did you make that choice?

Reply:
There's a KeyError, because existing_data also has the `id` key, for the flair template ID. Don't want that.

Reply:
[`.get()` never raises a `KeyError`, as per documentation](https://docs.python.org/3/library/stdtypes.html#dict.get). My question in the comment above is, in essence, "why didn't you do `data.get(key, None)`?" (I say "in essence" because the `None` can be omitted, as `None` is the default value).